llarp/crypto/encrypted_frame.cpp

```diff
@@ -7,42 +7,31 @@
 namespace llarp
 {
   bool
-  EncryptedFrame::EncryptInPlace(const SecretKey& ourSecretKey,
-                                 const PubKey& otherPubkey)
+  EncryptedFrame::DoEncrypt(const SharedSecret& shared, bool noDH)
   {
-    // format of frame is
-    // <32 bytes keyed hash of following data>
-    // <32 bytes nonce>
-    // <32 bytes pubkey>
-    // <N bytes encrypted payload>
-    //
     byte_t* hash     = data();
```

Comment:
You should give the whole codebase a once-over.  This is moved code, not new.  Can change though, I just kept it the same to minimize the amount of possible failure points.

Reply:
Probably should refactor all the places that take a raw `byte_t *` at the same time doing this sort of change, otherwise there's no gain that I can see.  Actually I'm not sure what it benefit this would have anyway other than looking fancy...

Reply:
Using `span` over raw pointer arithmetic has some reasonings [here](https://abseil.io/tips/93) and [here](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f24-use-a-spant-or-a-span_pt-to-designate-a-half-open-sequence)